Framework/Algorithms/src/PDCalibration.cpp

```diff
@@ -189,6 +195,10 @@ const std::string PDCalibration::summary() const {
 void PDCalibration::init() {
   declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>("InputWorkspace", "", Direction::Input),
                   "Input workspace containing spectra as a function of TOF measured on a standard sample.");
+  declareProperty(std::make_unique<WorkspaceProperty<MaskWorkspace>>("MaskWorkspace", "", Direction::Output),
```

Comment:
Small point, but extra properties should be added at the end so as not to break the positional arguments of the python callable automatically generated.

Reply:
Richard:  With respect to using the detector masks on the input workspaces, YES, that would definitely be preferable.  However, there are too many algorithms in Mantid which do _not_ do that.   I had started out these changes with that in mind, but I decided that a more robust interim solution involved using the explicit mask workspace.

Reply:
I will take a look at, and fix, the `declareProperty` order.

Reply:
The `MaskWorkspace` probably isn't in the right place, but it should stay in the "group" with input options
```cpp
  std::string inputGroup("Input Options");
  setPropertyGroup("InputWorkspace", inputGroup);
  setPropertyGroup("StartWorkspaceIndex", inputGroup);
  setPropertyGroup("StopWorkspaceIndex", inputGroup);
  setPropertyGroup("TofBinning", inputGroup);
  setPropertyGroup("PreviousCalibrationFile", inputGroup);
  setPropertyGroup("PreviousCalibrationTable", inputGroup);
```
Putting the new property near the end takes it away from its friends.

I think the vast majority of people using this algorithm are instrument scientists and should have been taught to call the function wrappers using kwargs rather than cargs.

Reply:
Yeh, I see your point. If an argument was added in 15th position of 20 then I'd say no one will be calling a function with 15 positional arguments, but in this case it's the second one!
I've thought for a while that it'd be nice to have the automatic GUI generation put arguments in the same property group together (even if one was added much later at the end) - then you get positional arguments order preserved and a sensible graphic! What do you think?

Reply:
That would be a very nice change, but outside of this PR

Reply:
I ended up leaving the `MaskWorkspace` property location the _same_ in `GetDetectorOffsets`, and moved it to _near_ (although not _at_) the end in `PDCalibration`.  **Please take a look and let me know if this still doesn't seem correct.**  This property is now defined as an `Output` with `PropertyMode::Optional`.  Mantid's python API does not pay correct attention to the latter, and at present it seems necessary to give all output parameters a name (thus the dummy name specified).   **Lastly, the parameters display did not look correct to me when the property was added to the "Input Options" group -- that's why it's still out on its own.**